dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -1537,6 +1540,93 @@ def test_profiler_factory_class_creates_correct_profiler(self, *mocks):
+            data = dp.Data(os.path.join(datapth, test_file))
+            save_profile = dp.StructuredProfiler(data)
+
+            # Save and Load profile with Mock IO
+            with mock.patch('builtins.open') as m:
+                mock_file = setup_save_mock_open(m)
+                save_profile.save()
+                mock_file.seek(0)
+                with mock.patch('dataprofiler.profilers.profile_builder.'
+                                'DataLabeler'):
+                    load_profile = dp.Profiler.load("mock.pkl")
+
+            # Check that reports are equivalent
+            save_report = _clean_report(save_profile.report())
+            load_report = _clean_report(load_profile.report())
+            self.assertDictEqual(save_report, load_report)
+
+    def test_save_and_load_unstructured(self):
+        data_folder = "dataprofiler/tests/data/"
+        test_files = ["txt/code.txt", "txt/sentence-10x.txt"]
+
+        for test_file in test_files:
+            # Create Data and StructuredProfiler objects
+            data = dp.Data(os.path.join(data_folder, test_file))
+            save_profile = UnstructuredProfiler(data)
+
+            # address case where empty_line_count is 0
+            save_profile._empty_line_count = 1
+
+            # store the expected data_labeler
+            data_labeler = save_profile.options.data_labeler.data_labeler_object
+
+            # Save and Load profile with Mock IO
+            with mock.patch('builtins.open') as m:
+                mock_file = setup_save_mock_open(m)
+                save_profile.save()
+
+                # make sure data_labeler unchanged
+                self.assertIs(
+                    data_labeler,
+                    save_profile.options.data_labeler.data_labeler_object)
+                self.assertIs(
+                    data_labeler,
+                    save_profile._profile._profiles[
+                        'data_labeler'].data_labeler)
+
+                mock_file.seek(0)
+                with mock.patch('dataprofiler.profilers.profile_builder.'
+                                'DataLabeler', return_value=data_labeler):
+                    load_profile = Profiler.load("mock.pkl")
+
+            # validate loaded profile has same data labeler class
+            self.assertIsInstance(
+                load_profile.options.data_labeler.data_labeler_object,
+                data_labeler.__class__)
+            self.assertIsInstance(
+                load_profile.profile._profiles['data_labeler'].data_labeler,
+                data_labeler.__class__)
+
+            # Check that reports are equivalent
+            save_report = save_profile.report()
+            load_report = load_profile.report()
+            self.assertDictEqual(save_report, load_report)
+
+            # validate both are still usable after
+            save_profile.update_profile(pd.DataFrame(['test']))
+            load_profile.update_profile(pd.DataFrame(['test']))
```

Comment:
Why aren't these checks in the structured test function?

Reply:
added data_label checks

Reply:
added these checks too